servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancerFactory.java

```diff
@@ -70,6 +91,10 @@ private RoundRobinLoadBalancerFactory(boolean eagerConnectionShutdown) {
      */
     public static final class Builder<ResolvedAddress, C extends LoadBalancedConnection> {
         private boolean eagerConnectionShutdown = EAGER_CONNECTION_SHUTDOWN_ENABLED;
+        @Nullable
+        private Executor backgroundExecutor;
+        private Duration healthCheckInterval = Duration.ZERO;
+        private int healthCheckFailedConnectionsThreshold;
```

Comment:
Instead of checking at the `build` time, you can initialize all defaults right here. Validation of input params will prevent misconfiguration.

Reply:
The idea her was to utilize lazy initialization as suggested by yourself in a previous comment. If I initialize the `backgroundExecutor` here, the whole point is missed. Were you thinking about something different though?

Reply:
It makes sense to make `backgroundExecutor` initialization lazy, in this comment I was referring only to `healthCheckInterval` and `healthCheckFailedConnectionsThreshold`. Sorry for confusion